### PR TITLE
github: run unit tests workflow

### LINKED ISSUE #47: github: add workflow to run the unit tests
There are already a couple of unit tests implemented in the project but they aren't executed on the current [build workflow](https://github.com/confidential-containers/cloud-api-adaptor/tree/staging/.github/workflows/build.yaml).

Let's extend the build workflow to all run the unit tests afterwards.

### PR DESCRIPTION
**Pull request by wainersm**
This adds a step on the build workflow in order to run the existing unit tests.

The tests for ibmcloud aren't compiling. I opened the issue #48 and disabled them all from CI.

The tests that need root privileges are refactored so that they use a common utility function for skipping. Even though running the CI pipeline as `sudo` some tests are failing to create network reosurces; those were also disabled from CI (see the issue #52).

I was working to get the test results nicely displayed on Github UI. I investigated the use of either https://github.com/marketplace/actions/junit-report-action or https://github.com/dorny/test-reporter but they both have limitations that didn't make me comfortable to enable them on the project. So for now they aren't used but I decided to keep the code that convert the raw output of the tests into JUnit XML with the hope that one day I could go back to this reporting problem.

All in all, at this point the CI pipelines should be all green but some unit tests are disabled.

Fixes #47

### CODE REVIEW

**Comment by wainersm** [comment]
Updated this PR to fix a commit that was missing the SOB. Now DCO bot is happy.

**Comment by wainersm** [comment]
Cc @bpradipt @stevenhorsman

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -19,6 +17,10 @@ jobs:
           - aws
           - ibmcloud
           - libvirt
+        runner:
+          - ubuntu-latest
```
Comment (stevenhorsman) [inline .github/workflows/build.yaml:21]: Just a thought, but I wonder if hooking this into the `runs-on` field e.g. `runs-on: ${{ matrix.runner }}`, might be more intuitive if we decided to add support for more OSs in future, and would remove a duplication of `ubuntu-latest`

**Review by stevenhorsman — APPROVED** [review]
@wainersm - I made an optional, minor comment, but this looks good. Thanks for raising #48 - I'll add it to the TODO list!

File: .github/workflows/build.yaml (hunk 1 above)
Comment (wainersm) [reply .github/workflows/build.yaml:21]: It does make sense. I will change that.

**Review by bpradipt — APPROVED** [review]
/lgtm
Thanks @wainersm

**Pull request merged**